x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/90_semantic_text_source_filtering.yml

```diff
@@ -0,0 +1,88 @@
+setup:
+  - requires:
+      cluster_features: "gte_v8.15.0"
+      reason: semantic_text introduced in 8.15.0
+
+  - do:
+      indices.create:
+        index: test-index
+        body:
+          mappings:
+            properties:
+              sparse_field:
+                type: semantic_text
+                inference_id: sparse-inference-id
```

Comment:
You should create the `sparse-inference-id` inference endpoint first:

```yaml
  - do:
      inference.put:
        task_type: sparse_embedding
        inference_id: sparse-inference-id
        body: >
          {
            "service": "test_service",
            "service_settings": {
              "model": "my_model",
              "api_key": "abc64"
            },
            "task_settings": {
            }
          }
```

Reply:
Yeah, I have seen the example but got confused because during `field_caps` we did not need to do that. I followed the tests defined [10_semantic_text_field_mapping](https://github.com/elastic/elasticsearch/blob/fa2e2812d4dd34a1ccf18009b589e72d5638a908/x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/10_semantic_text_field_mapping.yml#L6-L17)

So the `10_semantic_text_field_mapping` assumed that it generated embedding whereas we need the inference API call to generate the embeddings?

Reply:
What you wrote works, but only because you manually created the embeddings in the expected format. It will be friendlier to future devs if you use an inference endpoint to automatically generate the embeddings in the correct format.

`10_semantic_text_field_mapping` tests didn't create an inference endpoint because those tests were specifically about testing the mappings. Thus, manually specifying the internal mappings was part of the test. In our case, we are not testing the internal mappings, so we should not manually specify them. Doing so makes these tests more brittle and harder to maintain.